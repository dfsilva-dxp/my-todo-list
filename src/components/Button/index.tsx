import { forwardRef, ForwardRefRenderFunction } from "react";

import * as S from "./styles";
import { ButtonProps } from "./types";

const Button: ForwardRefRenderFunction<S.WrapperProps, ButtonProps> = (
  {
    children,
    size = "medium",
    fullWidth = false,
    minimal = false,
    icon,
    ...props
  },
  ref
) => {
  return (
    <S.Wrapper
      size={size}
      fullWidth={fullWidth}
      hasIcon={!!icon}
      minimal={minimal}
      ref={ref}
      {...props}
    >
      {!!children && <span>{children}</span>}
      {!!icon && icon}
    </S.Wrapper>
  );
};

export default forwardRef(Button);
