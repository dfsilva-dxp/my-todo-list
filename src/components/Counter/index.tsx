import * as S from "./styles";

import { ICounterProps } from "./types";

const Counter = ({ value }: ICounterProps) => {
  return (
    <S.Wrapper role="badge" aria-label="counter">
      {value}
    </S.Wrapper>
  );
};

export default Counter;
