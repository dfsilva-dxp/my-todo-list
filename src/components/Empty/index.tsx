import clipboardSVG from "@assets/clipboard.svg";

import * as S from "./styles";

const Empty = () => {
  return (
    <S.Wrapper>
      <img
        src={clipboardSVG}
        alt="clipboard icon"
        title="clipboard icon"
        aria-label="clipboard icon"
      />

      <S.TextEmptyWrapper>
        <h4>Você ainda não tem tarefas cadastradas</h4>
        <p>Crie tarefas e organize seus itens a fazer</p>
      </S.TextEmptyWrapper>
    </S.Wrapper>
  );
};

export default Empty;
