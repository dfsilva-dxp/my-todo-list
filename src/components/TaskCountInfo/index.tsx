import * as S from "./styles";

import { ITaskCountInfoProps } from "./types";

const TaskCountInfo = ({
  children,
  badge,
  variant = "created"
}: ITaskCountInfoProps) => {
  return (
    <S.Wrapper variant={variant} role="info" aria-label="count info">
      <p>{children}</p>
      {badge}
    </S.Wrapper>
  );
};

export default TaskCountInfo;
