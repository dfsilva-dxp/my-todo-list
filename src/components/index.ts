export { default as Button } from "./Button";
export { default as Checkbox } from "./Checkbox";
export { default as Container } from "./Container";
export { default as Counter } from "./Counter";
export { default as Empty } from "./Empty";
export { default as Form } from "./Form";
export { default as Header } from "./Header";
export { default as Input } from "./Input";
export { default as Logo } from "./Logo";
export { default as Task } from "./Task";
export { default as TaskCountInfo } from "./TaskCountInfo";
export { default as TaskList } from "./TaskList";
