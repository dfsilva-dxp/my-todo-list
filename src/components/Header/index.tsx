import { Logo } from "@/components";

import * as S from "./styles";

import logoSVG from "@/assets/logo.svg";

const Header = () => {
  return (
    <S.Wrapper role="banner">
      <Logo
        src={logoSVG}
        title="Logo Todo App"
        alt="Logo Todo App"
        aria-label="logo"
      />
    </S.Wrapper>
  );
};

export default Header;
