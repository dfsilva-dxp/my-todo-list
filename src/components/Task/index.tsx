import { useState } from "react";
import { Trash } from "phosphor-react";

import { Button, Checkbox } from "@/components";

import useTaskContext from "@/hooks/useTasksContext";

import * as S from "./styles";

import { ITaskProps } from "./types";

const Task = ({ description, checked, taskId }: ITaskProps) => {
  const [isChecked, setIsChecked] = useState(checked);

  const { updateTask } = useTaskContext();

  const onCheck = (status: boolean) => {
    setIsChecked(status);
    updateTask(taskId, status);
  };

  return (
    <S.Wrapper role="listitem" aria-label="task card">
      <Checkbox isChecked={isChecked} onCheck={onCheck} />

      <S.TaskDescription status={isChecked ? "done" : "created"}>
        {description}
      </S.TaskDescription>

      <Button
        type="button"
        minimal
        size="small"
        icon={<Trash size={20} weight="bold" />}
      />
    </S.Wrapper>
  );
};

export default Task;
