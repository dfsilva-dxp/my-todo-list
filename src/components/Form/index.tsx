import { FormEvent, useState } from "react";
import { PlusCircle } from "phosphor-react";

import { Button, TextField } from "@/components";

import useTaskContext from "@/hooks/useTasksContext";

import * as S from "./styles";

import { ITask } from "@components/Task/types";

const Form = () => {
  const [inputValue, setInputValue] = useState("");

  const { createTask } = useTaskContext();

  const onInputChange = (value: string) => {
    setInputValue(value);
  };

  const handleSubmit = (evt: FormEvent<HTMLFormElement>) => {
    evt.preventDefault();
    const id = Math.floor(Date.now() * Math.random());

    const data: ITask = {
      id: String(id),
      description: inputValue,
      status: "created",
      checked: false,
      createdAt: new Date()
    };

    createTask(data);
    setInputValue("");
  };

  return (
    <S.Wrapper onSubmit={handleSubmit}>
      <TextField
        placeholder="Adicione uma nova tarefa"
        initialValue={inputValue}
        onInputValue={onInputChange}
      />
      <Button
        type="submit"
        size="small"
        icon={<PlusCircle size={20} weight="bold" />}
      >
        Criar
      </Button>
    </S.Wrapper>
  );
};

export default Form;
