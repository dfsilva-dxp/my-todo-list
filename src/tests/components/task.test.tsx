import { renderWithTheme } from "@/utils/tests/renderWithTheme";
import { screen } from "@testing-library/react";

import { Task } from "@/components";

describe("Task", () => {
  it("should render currectly", () => {
    renderWithTheme(
      <Task
        description="Estudar ReactJS - Hooks e Custom Hooks"
        status="created"
      />
    );

    expect(screen.getByRole("checkbox")).toBeInTheDocument();
    expect(
      screen.getByText("Estudar ReactJS - Hooks e Custom Hooks")
    ).toBeInTheDocument();
    expect(screen.getByRole("button")).toBeInTheDocument();
  });
  it("should render the task with status created", () => {
    const { container } = renderWithTheme(
      <Task
        description="Estudar ReactJS - Hooks e Custom Hooks"
        status="created"
      />
    );

    expect(screen.getByRole("checkbox")).not.toBeChecked();
    expect(
      screen.getByText("Estudar ReactJS - Hooks e Custom Hooks")
    ).toHaveStyle({
      color: "#F2F2F2",
      "text-decoration": "none"
    });

    expect(container).toMatchSnapshot();
  });

  it("should render the task with status done", () => {
    renderWithTheme(
      <Task
        description="Estudar ReactJS - Hooks e Custom Hooks"
        status="done"
      />
    );

    expect(screen.getByRole("checkbox")).toBeChecked();
    expect(
      screen.getByText("Estudar ReactJS - Hooks e Custom Hooks")
    ).toHaveStyle({
      color: "#808080",
      "text-decoration": "line-through"
    });
  });
});
