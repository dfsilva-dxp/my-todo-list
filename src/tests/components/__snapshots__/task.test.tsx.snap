// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task should render the task with status created 1`] = `
.c4 {
  background: #4EA8DE;
  padding: 0.5rem;
  color: #F2F2F2;
  font-weight: 700;
  border: 0;
  border-radius: 0.5rem;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.5rem;
  height: 3rem;
  padding: 0.8rem 1rem;
  font-size: 0.875rem;
  background: none;
}

.c4:hover {
  background: #808080;
}

.c4 svg {
  width: 1.5rem;
}

.c4:hover {
  background: #0D0D0D;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c2 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 1.5rem;
  height: 1.5rem;
  border: 0.2rem solid #4EA8DE;
  border-radius: 100%;
  -webkit-transition: background border 0.15s ease-in-out;
  transition: background border 0.15s ease-in-out;
  position: relative;
  outline: none;
}

.c2:before {
  content: "";
  width: 0.3rem;
  height: 0.6rem;
  border: 0.15rem solid #F2F2F2;
  border-top: 0;
  border-left: 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  position: absolute;
  top: 0.1rem;
  right: 0.3rem;
  opacity: 0;
  -webkit-transition: 0.15s ease-in-out;
  transition: 0.15s ease-in-out;
}

.c2:hover {
  border-color: #1E6F9F;
  background: #262626;
}

.c2:checked {
  border-color: #5E60CE;
  background: #5E60CE;
}

.c2:checked:before {
  opacity: 1;
}

.c2:checked:hover {
  border-color: #9164fa;
  background: #9164fa;
}

.c0 {
  background: #262626;
  padding: 1rem;
  border: 1px solid #333333;
  border-radius: 0.5rem;
  box-shadow: 0 2px 8px 0 rgba(0,0,0,0.06);
  display: grid;
  grid-template-columns: auto 2fr auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c0 button:hover {
  color: #E25858;
}

.c3 {
  font-size: 1rem;
  line-height: 140%;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 2;
  display: -webkit-box;
  overflow: hidden;
  color: #F2F2F2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <section
    aria-label="task card"
    class="c0"
    role="listitem"
  >
    <div
      class="c1"
    >
      <input
        class="c2"
        id=""
        role="checkbox"
        type="checkbox"
        value=""
      />
    </div>
    <p
      class="c3"
    >
      Estudar ReactJS - Hooks e Custom Hooks
    </p>
    <button
      class="c4"
      type="button"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 256 256"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="216"
          x2="40"
          y1="60"
          y2="60"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="104"
          x2="104"
          y1="104"
          y2="168"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
          x1="152"
          x2="152"
          y1="104"
          y2="168"
        />
        <path
          d="M200,60V208a8,8,0,0,1-8,8H64a8,8,0,0,1-8-8V60"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
        />
        <path
          d="M168,60V36a16,16,0,0,0-16-16H104A16,16,0,0,0,88,36V60"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="24"
        />
      </svg>
    </button>
  </section>
</div>
`;
